pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -708,8 +712,9 @@ func (c *Controller) serviceInstancesFromWorkloadInstances(svc *model.Service, r
 	workloadInstancesExist = len(c.workloadInstancesByIP) > 0
 	c.RUnlock()
 
+	// Only select internal Kubernetes services with selectors
```

Comment:
nit: This comment and code below is little confusing. It took a while for me to understand we are doing an exclusion here rather than selecting